make release failing sometimes

## Bug Report

Currently the `make release` is failing at the package stage with the error `cp: cannot stat '/tmp/tce-release/tanzu-framework/artifacts/linux/amd64/cli/core/v0.7.0/tanzu-core-linux_amd64': No such file or directory`

## Expected Behavior

`make release` to work with no errors

## Steps to Reproduce the Bug

Run `make release`

## Debugging Details and Root Cause

On debugging the issue I found out that the core binary is not getting built as part of the Tanzu Framework build and hence the error at the package stage that there's no binary found

core binary is usually built as part of the `make build-install-cli-all` and there's usually a log that says `building core binary`. This didn't happen though, especially in the CI/CD environments. This was noticed in a downstream pipeline and I also noticed it when I tried the `make release` in a brand new EC2 Linux machine

It worked in my local in Mac and it also worked the second time and other times I tried to do a `make release` in the brand new EC2 Linux machine

On digging through the logs there were also errors that was noticed which we weren't sure if it's related to the issue - one was #1176 , another was

```
...
...
✖  🦁 - error unmarshalling plugin descriptor: invalid character 'g' looking for beginning of value
...
...
✖  🦁 - building plugin "cmd/cli/plugin-admin/codegen" failed - invalid character 'g' looking for beginning of value
exit status 1
make[1]: *** [Makefile:233: build-plugin-admin] Error 1
```

On digging through I noticed that the `make build-install-cli-all` in turn calls `build-plugin-admin` target among other things and since `build-plugin-admin` failed, it didn't proceed to the remaining targets it was supposed to run, for example `build-cli-linux-amd64`, `build-cli-windows-amd64` and `build-cli-darwin-amd64`, which in turn build the core binary

So, basically the make target failed silently half way in between and proceeded to other build script commands

To check why and how the `build-plugin-admin` target was failing while build codegen plugin I had added extra logs to see what the plugin descriptor was. Looks like an `info` command is used to print the JSON plugin descriptor and the error I noticed was this -

```
ℹ  🦁 - building plugin at path "cmd/cli/plugin-admin/codegen"
....
  🦁 - error unmarshalling plugin descriptor: invalid character 'g' looking for beginning of value . Plugin descriptor - go: downloading sigs.k8s.io/controller-tools v0.3.0
go: downloading golang.org/x/tools v0.1.0
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
{"name":"codegen","description":"Tanzu code generation tool","version":"v1.4.0-dev.0","buildSHA":"4395741","group":"Admin","docURL":"","completionType":0}
```

You can find it here - https://github.com/karuppiah7890/tce/runs/3179489837?check_suite_focus=true

So, basically you can see that it's downloading some golang modules while running the `info` command using `go run` internally - related code - https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/builder/command/cli.go#L299-L333

I think the reason why it happens in CI/CD only is because in CI/CD the machines are brand new and the first time `make release` runs there are no golang modules in it and hence all the downloading and hence the above happens. But the next time it runs, it will pass, as the golang modules will have already been downloaded. This is one reason why it was not easy to debug because all of the developer machines already had the necessary golang modules and the issue occurred only in CI/CD pipelines or brand new machines

This is something we can talk about to the Tanzu Framework team and understand why this is happening and how it can be avoided and what they are doing in their CI/CD pipelines and if this is / isn't affecting them etc. I can see two possible things causing the golang module download - one is the `go run` and other is the code that calls `go mod download` - from here https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/builder/command/cli.go#L306-L315 , specifically - https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/builder/command/cli.go#L310 - which in turn calls - https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/builder/command/cli.go#L561-L583 which does `go mod download`. Also it uses the `CombinedOutput` of the `info` command run https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/builder/command/cli.go#L321 which gives all the output - standard output and standard error which can get tricky

Apart from discussing with Tanzu Framework team for a long term fix or an idea on how to avoid this, as a work around, I propose we try to run a `go mod download` and `go mod tidy` as part of the Tanzu Framework build just before we run any `make` commands

I'll raise a PR for this work around

--- a/hack/build-tanzu.sh
+++ b/hack/build-tanzu.sh
@@ -53,6 +53,9 @@ BUILD_SHA="$(git describe --match="$(git rev-parse --short HEAD)" --always)"
 sed -i.bak -e "s/ --dirty//g" ./Makefile && rm ./Makefile.bak
 sed -i.bak -e "s/\$(shell git describe --tags --abbrev=0 2>\$(NUL))/${FRAMEWORK_BUILD_VERSION}/g" ./Makefile && rm ./Makefile.bak
 
+go mod download
+go mod tidy
+
 # allow unstable (non-GA) version plugins
 if [[ "${TCE_BUILD_VERSION}" == *"-"* ]]; then
 make controller-gen

